homeassistant/components/vegehub/sensor.py

```diff
@@ -0,0 +1,100 @@
+"""Sensor configuration for VegeHub integration."""
+
+from itertools import count
+
+from homeassistant.components.sensor import (
+    SensorDeviceClass,
+    SensorEntity,
+    SensorEntityDescription,
+)
+from homeassistant.const import UnitOfElectricPotential
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.entity_platform import AddConfigEntryEntitiesCallback
+
+from .coordinator import VegeHubConfigEntry, VegeHubCoordinator
+from .entity import VegeHubEntity
+
+SENSOR_TYPES: dict[str, SensorEntityDescription] = {
+    "analog_sensor": SensorEntityDescription(
+        key="analog_sensor",
+        translation_key="analog_sensor",
+        device_class=SensorDeviceClass.VOLTAGE,
+        native_unit_of_measurement=UnitOfElectricPotential.VOLT,
+        suggested_display_precision=2,
+    ),
+    "battery": SensorEntityDescription(
+        key="battery",
+        translation_key="battery",
+        device_class=SensorDeviceClass.VOLTAGE,
+        native_unit_of_measurement=UnitOfElectricPotential.VOLT,
+        suggested_display_precision=1,
+    ),
+}
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    config_entry: VegeHubConfigEntry,
+    async_add_entities: AddConfigEntryEntitiesCallback,
+) -> None:
+    """Set up Vegetronix sensors from a config entry."""
+    sensors: list[VegeHubSensor] = []
+    coordinator = config_entry.runtime_data
+
+    # This is the index in the updates from the VegeHub that will correspond to
+    # each sensor.
+    update_index = count(0)
+
+    # Add each analog sensor input
+    for _i in range(coordinator.vegehub.num_sensors):
+        sensor = VegeHubSensor(
+            index=next(update_index),
+            coordinator=coordinator,
+            description=SENSOR_TYPES["analog_sensor"],
+        )
+        sensors.append(sensor)
+
+    # Add the battery sensor
+    if not coordinator.vegehub.is_ac:
+        sensors.append(
+            VegeHubSensor(
+                index=next(update_index),
+                coordinator=coordinator,
+                description=SENSOR_TYPES["battery"],
+            )
+        )
+
+    async_add_entities(sensors)
+
+
+class VegeHubSensor(VegeHubEntity, SensorEntity):
+    """Class for VegeHub Analog Sensors."""
+
+    _attr_device_class = SensorDeviceClass.VOLTAGE
+    _attr_native_unit_of_measurement = UnitOfElectricPotential.VOLT
+    _attr_suggested_display_precision = 2
```

Comment:
why do we have device class, native uom and precision here if we already have this defined in the entity description?